cmake/deprecated.cmake

```diff
@@ -0,0 +1,42 @@
+#######
+# Check for deprecated arguments, provide warning, and set the
+# corresponding option.
+#
+# This file is being added in support of https://github.com/Unidata/netcdf-c/pull/2895 and may eventually
+# no longer be necessary and removed in the future. 
+# 
+#######
+
+set(DEPR_OPT "Warning! Deprecated Options used. Please migrate your build system as follows:" CACHE INTERNAL "" FORCE)
+
+function(check_depr_opt arg)
+
+    if(DEFINED ${arg})
```

Comment:
`if (DEFINED CACHE{${arg}})` doesn't hurt to check more narrowly for `-D` options, as opposed to local variable.